### ISSUE TITLE
grpc traffic stealing doesn't work with istio

### ISSUE THREAD

**Issue by jboykin-bread** [metalbear-co/mirrord#2794]
### Bug Description

Hey folks! Hope y'all are doing well 😄

I think I've found a bug with mirrord traffic stealing and istio v1.23. First encountered it in our AWS EKS clusters, but I'm also able to reproduce it with a local kind cluster and a simple http service in Go. 

When running `mirrord exec -e --steal --target pod/my-service -- ./my-service` with istio injection DISABLED, everything works as expected. I can hit my-service:8000 in the cluster and see that it's stolen and routed to my local process.

When running the same mirrord command with istio injection ENABLED in my namespace on all pods, my traffic to my-service:8000  is no longer stolen and routed to my local process. It either times out client-side, or is routed to the currently running Pod instead of my local process.

Happy to provide more details if necessary 😃 I wrote up detailed steps on how to reproduce in the readme here -> https://github.com/jboykin-bread/mirrord-traffic-stealing-reproduction

### Steps to Reproduce

Wrote up detailed steps on how to reproduce in the readme here -> https://github.com/jboykin-bread/mirrord-traffic-stealing-reproduction

### mirrord layer logs

Agent logs
```shell
❯ kubectl logs -f color-server-77ff58464f-nnfz8 -c mirrord-agent-n3ulshz1tz
agent ready - version 3.118.0
2024-10-01T21:14:13.231613Z  WARN ThreadId(07) mirrord_agent::outgoing::udp: interceptor_task -> no messages left
2024-10-01T21:19:48.239231Z  WARN ThreadId(10) mirrord_agent::outgoing::udp: interceptor_task -> no messages left
2024-10-01T21:19:48.247636Z  INFO ThreadId(01) mirrord_agent::entrypoint: main -> mirrord-agent `start` exiting successfully.
2024-10-01T21:19:48.271699Z  INFO ThreadId(01) mirrord_agent::entrypoint: main -> mirrord-agent `start` exiting successfully.
```

### Internal proxy logs

```shell
❯ cat /private/tmp/mirrord-intproxy-1727817252-32LQ3W7
2024-10-01T21:14:12.241194Z  INFO mirrord_intproxy::layer_initializer: new session process_info=ProcessInfo { pid: 75133, name: "color-server", cmdline: ["./color-server", "-color=green"], loaded: true }
2024-10-01T21:19:47.392781Z  WARN mirrord_kube::api::kubernetes::portforwarder: fininsed retry strategy closing connection connect_info=AgentKubernetesConnectInfo { pod_name: "color-server-77ff58464f-nnfz8", agent_port: 50781, namespace: Some("color-server"), agent_version: Some("3.118.0") }
```

### Your operating system and version

macos 14.5 

### Local process

color-server: Mach-O 64-bit executable arm64

**Comment by aviramha** [comment]
Thanks for reporting this!<br>Was there any trigger that caused it to start happening? was it an upgrade to istio 1.23 from a lower vesion?

**Comment by aviramha** [comment]
Thanks for the great repro steps. We'll try to prioritize it soon!

**Comment by aviramha** [comment]
Managed to reproduce this using user's example, thank you very much.<br>In my case, first request is stolen, but then fails like mentioned.

**Comment by aviramha** [comment]
[https://github.com/metalbear-co/mirrord/issues/2456](https://github.com/metalbear-co/mirrord/issues/2456)

Managed to fix it by applying same fix - `echo 1 > /proc/sys/net/ipv4/conf/all/route_localnet`

**Comment by aviramha** [comment]
False alarm, it did make it work but then I restarted the mirrord session and got back to same issue ;(

**Comment by aviramha** [comment]
Seems to be caused by `iptables -I OUTPUT -j DROP -p tcp -m tcp --tcpflags RST` of VPN.

**Comment by aviramha** [comment]
Fixed in 3.119.0
